core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -63,13 +65,23 @@ public static <T> CloseableIterable<T> filter(CloseableIterable<T> rows, Functio
     return equalityFilter.filter(rows);
   }
 
-  public static <T> CloseableIterable<T> filter(CloseableIterable<T> rows, Function<T, Long> rowToPosition,
-                                                PositionDeleteIndex deleteSet) {
-    if (deleteSet.isEmpty()) {
-      return rows;
-    }
+  public static <T> CloseableIterable<T> markDeleted(CloseableIterable<T> rows, Predicate<T> isDeleted,
```

Comment:
I think this API does a lot of things, can it be like:
``` CloseableIterable<T> markDeleted(CloseableIterable<T> rows, Consumer<T> deleteMarker) ```

 just markDeleted and then have the user call:
``` Deletes.markDeleted(filtered, marker)```

Reply:
Not sure I understand you correctly. We cannot filter anything here, right? We keep all rows while mark the deleted rows.